// grunt-contrib-lefdupload/lib/upload-moudle.js

var fs = require('fs'),
    needle = require('needle');
var requestOptions,postData,url;
exports.setOptions = function(upload_conf) {
    url = upload_conf.url;
    postData = upload_conf.postData;
    requestOptions= upload_conf.request;
}


function getFileName(file) {
    file = file.split(/\/|\\/);
    file = file[file.length - 1];
    return file;
}

function needleUpload(file, callback) {
    postData.single_upload_file = {
        buffer: fs.readFileSync(file),
        filename: getFileName(file),
        content_type: 'application/octet-stream'
    };
    needle.post(url, postData, requestOptions, function(err, resp, body) {
        var json;
        if (err) {
            throw err;
            return;
        }

        try {
            if(typeof body == 'string') {
                json = JSON.parse(body);
            } else {
                json = body;
            }
        } catch (ex) {
            callback('error content：' + ex);
            return;
        }
        if (json.state == 1) {
            callback(false, json.file);
        } else if(json.state == 4){
            callback('auth error'); 
        } else {
            callback('error content：' + body.msg);
        }
    });
}

//上传单文件
exports.uploadOneFile = function(folder, file, callback) {
    postData.fdir = folder;
    needleUpload(file, function(err, fileurl) {
        postData.single_upload_file = null;
        if (err) {
            callback(file + ':' + err);
        } else {
            callback(false, fileurl,file);
        }
    });
}
